File: merge.py
class Solution:#time O(m+n) space O(1)
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """

        p=len(nums1)-1
        m=m-1
        n=n-1
        while (m>=0 and n>=0):
            if nums1[m]>=nums2[n]:
                nums1[p]=nums1[m]
                m-=1
            else:
                nums1[p]=nums2[n]
                n-=1
            p-=1
        while n>=0:
            nums1[p]=nums2[n]
            p-=1
            n-=1

File: merge.js
// Time Complexity : O(m+n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes

// Comparing value between m-1 and n-1 index and the value which is getting stored in nums1 array at a index.

/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    if(nums1 === null) retrun;
    let p1 = m - 1;
    let p2 = n - 1;
    let idx = m+n - 1;
    while(p1 >= 0 && p2 >= 0){
        if(nums1[p1] > nums2[p2]) {
            nums1[idx] = nums1[p1];
            p1--;
        } else {
            nums1[idx] = nums2[p2];
            p2--;
        }
        idx--;
    } 
    while(p2 >= 0) {
        nums1[idx] = nums2[p2];
        p2--;
        idx--;
    }
};